modules/js-and-css/partytown-web-worker/load.php

```diff
@@ -0,0 +1,139 @@
+<?php
+/**
+ * Module Name: Partytown Web Worker
+ * Description: Add support for running JavaScript in a separate Web Worker with the help of the Partytown library.
+ * Experimental: Yes
+ *
+ * @since   n.e.x.t
+ * @package performance-lab
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * PartyTown Configuration
+ *
+ * @since n.e.x.t
+ * @see https://partytown.builder.io/configuration
+ * @return array
+ */
+function perflab_partytown_web_worker_configuration() {
+	$plugin_dir           = plugin_dir_path( __FILE__ );
+	$content_dir_basename = basename( WP_CONTENT_DIR );
+
+	$config = array(
+		'lib'     => substr( $plugin_dir, strpos( $plugin_dir, $content_dir_basename ) - 1 ) . 'assets/js/partytown/',
+		'forward' => array(),
+	);
+
+	/**
+	 * Add configuration for PartyTown.
+	 *
+	 * @since n.e.x.t
+	 * @see <https://partytown.builder.io/configuration>.
+	 * @param array $config Configuration for PartyTown.
+	 * @return array
+	 */
+	return apply_filters( 'perflab_partytown_configuration', $config );
+}
+
+/**
+ * Initialize PartyTown
+ *
+ * @since n.e.x.t
+ * @return void
+ */
+function perflab_partytown_web_worker_init() {
+	$partytown_js = file_get_contents( __DIR__ . '/assets/js/partytown/partytown.js' ); // phpcs:ignore WordPress.WP.AlternativeFunctions.file_get_contents_file_get_contents -- It's a local filesystem path not a remote request.
+
+	wp_register_script(
+		'partytown',
+		'',
+		array(),
+		PERFLAB_VERSION,
+		array( 'in_footer' => false )
+	);
+
+	wp_add_inline_script(
+		'partytown',
+		sprintf(
+			'window.partytown = %s;',
+			wp_json_encode( perflab_partytown_web_worker_configuration() )
+		),
+		'before'
+	);
+
+	wp_add_inline_script(
+		'partytown',
+		$partytown_js,
+		'after'
+	);
+}
+add_action( 'wp_enqueue_scripts', 'perflab_partytown_web_worker_init' );
+
+/**
+ * Helper function to get all scripts tags which has `partytown` dependency.
+ *
+ * @since n.e.x.t
+ *
+ * @return array Array of script handles with `partytown` dependency.
+ */
+function perflab_get_partytown_handles() {
+	global $wp_scripts;
+
+	$partytown_handles = array();
+	foreach ( $wp_scripts->registered as $handle => $script ) {
+		if ( ! empty( $script->deps ) && in_array( 'partytown', $script->deps, true ) ) {
```

Comment:
Correct me if I'm wrong, but let's say you have an existing script in a theme like:

```php
wp_enqueue_script( 'foo', 'https://example.com/foo.js' );
```

In order to opt-in this script to use Partytown, you'd have to do this if you wanted to opt-in from a separate plugin:

```php
wp_scripts()->registered['foo']->deps[] = 'partytown';
```

Is that right?

Reply:
Yup, that should do the trick.